src/x509.rs

```diff
@@ -79,6 +79,14 @@ pub struct AlgorithmIdentifier<'a> {
 #[derive(Debug, PartialEq)]
 pub struct X509Name<'a> {
     pub rdn_seq: Vec<RelativeDistinguishedName<'a>>,
+    pub(crate) raw: &'a [u8],
+}
+
+impl<'a> X509Name<'a> {
+    // Not using the AsRef trait, as that would not give back the full 'a lifetime
+    pub fn as_raw(&self) -> &'a [u8] {
```

Comment:
- I'm also curious about the comment on lifetimes (which seems similar to the above one)
- Could you share the use case where you needed the raw bytes? I'm asking mostly because, if this is useful for several fields, the `as_raw` method could be added to all parsed components

Reply:
Similar to the above; the AsRef trait takes no lifetime parameter, and the as_ref function can only return data with the lifetime on the X509Name; if the parsing is done in an inner function, the inner function won't be able to return slices from that.

My use case is sending OCSP requests and validating the reponses; the raw data is sometimes hashed (building the CertID for the OCSP protocol), it can also be copied (some parts of the request should exactly match the certificates that are being checked), and compared (to check that the response really matches the request).